dashboard/15-final/app/api/auth/[...nextauth]/route.ts

```diff
@@ -0,0 +1,53 @@
+import NextAuth, { NextAuthOptions } from 'next-auth';
+import CredentialsProvider from 'next-auth/providers/credentials';
+import bcrypt from 'bcrypt';
+import { fetchUser } from '../../../lib/data';
+import { User } from '@/app/lib/definitions';
+
+const authOptions: NextAuthOptions = {
+  providers: [
+    CredentialsProvider({
+      name: 'credentials',
+      credentials: {},
+
+      // TODO: Not sure how to type authorize function correctly
+      // @ts-ignore
+      async authorize(credentials: {
+        email: string;
+        password: string;
+      }): Promise<User | null | undefined> {
+        const { email, password } = credentials;
+
+        try {
+          const user = await fetchUser(email);
```

Comment:
Calling this `fetchUser` will give the wrong idea. A common mistake is that people create an API route and then `fetch` it, instead of importing the logic. I can see that `lib/data` is a function not an API route, but I would prefer if we just had that logic directly here or call it something else than `fetchUser`. Maybe `getUser`.